### PR TITLE
Disable state transition stats verification in rebalancer cluster integration test

### PR DESCRIPTION
**Pull request by siddharthteotia**
Looks like state transition stats (from ONLINE to OFFLINE and OFFLINE to DROPPED) are potentially flakey as they are reliant on timing.

These stats are reliant on the fact that state transition has happened (callback invoked on the segment server) for the server that lost a segment as part of rebalancing.

While the rebalancer waits for all the ideal state changes to get reflected in external view, we don't yet check if the servers have indeed dropped the segments. So it is possible that server may not have done so by the time test finishes and thus we sometimes notice the segment dropped stats off by 1.

Disabling the stat verification to keep tests happy and until we make the external view check stronger by ensuring that it doesn't have any extra servers for a segment indicating that servers that were supposed to drop segments as part of rebalancing have truly done so. 

Will do this in a follow-up PR soon after but currently disabling to keep build clean.

### CODE REVIEW

**Comment by siddharthteotia** [comment]
cc @jackjlli @mcvsubbu

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TableRebalancerAdminToolClusterIntegrationTest.java [hunk 1]
```diff
@@ -67,6 +67,7 @@
   private static final int NUM_INITIAL_SERVERS = 3;
   private SegmentStateTransitionStats _segmentStateTransitionStats;
   private boolean _raiseErrorOnSegmentStateTransition = false;
+  private final boolean _checkStateTransitionStats = false;
```
Comment (Jackie-Jiang) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TableRebalancerAdminToolClusterIntegrationTest.java]: Don't add this boolean, just comment out the lines and add TODO to add them back in the next pr

**Comment by jackjlli** [comment]
Agree with @Jackie-Jiang that you can just comment out the assertions.

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TableRebalancerAdminToolClusterIntegrationTest.java (hunk 1 above)
Comment (siddharthteotia) [reply pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TableRebalancerAdminToolClusterIntegrationTest.java]: Done

**Comment by siddharthteotia** [comment]
> Agree with @Jackie-Jiang that you can just comment out the assertions.

Done

File: pinot-common/src/main/java/org/apache/pinot/common/function/AggregationFunctionType.java [hunk 2]
```diff
@@ -48,7 +48,8 @@
   DISTINCTCOUNTRAWHLLMV("distinctCountRawHLLMV"),
   PERCENTILEMV("percentileMV"),
   PERCENTILEESTMV("percentileEstMV"),
-  PERCENTILETDIGESTMV("percentileTDigestMV");
+  PERCENTILETDIGESTMV("percentileTDigestMV"),
+  DISTINCT("distinctValues");
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/function/AggregationFunctionType.java]: Wrong merge?

File: pinot-common/src/main/java/org/apache/pinot/pql/parsers/pql2/ast/FunctionCallAstNode.java [hunk 3]
```diff
@@ -82,8 +82,8 @@ public AggregationInfo buildAggregationInfo() {
       expression = "*";
     } else {
       List<? extends AstNode> children = getChildren();
-      if (children == null || children.size() != 1) {
-        throw new Pql2CompilationException("Aggregation function expects exact 1 argument");
+      if (children == null || (children.size() != 1 && !_name.equalsIgnoreCase("distinct"))) {
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/pql/parsers/pql2/ast/FunctionCallAstNode.java]: Same here?

File: pinot-controller/src/test/java/org/apache/pinot/controller/helix/PinotControllerModeTest.java [hunk 4]
```diff
@@ -56,7 +56,7 @@ public void testHelixOnlyController() {
     stopController();
   }
 
-  @Test
+  @Test(enabled = false)
```
Comment (Jackie-Jiang) [inline pinot-controller/src/test/java/org/apache/pinot/controller/helix/PinotControllerModeTest.java]: Don't disable this test, we need to fix it

Comment (jackjlli) [reply pinot-controller/src/test/java/org/apache/pinot/controller/helix/PinotControllerModeTest.java]: The test will be refactored in this PR:https://github.com/apache/incubator-pinot/pull/4323
We don't need to enable/disable resource from ideal state any more.
We'll re-enable the test in that PR.

File: pinot-common/src/main/java/org/apache/pinot/pql/parsers/pql2/ast/FunctionCallAstNode.java (hunk 3 above)
Comment (siddharthteotia) [reply pinot-common/src/main/java/org/apache/pinot/pql/parsers/pql2/ast/FunctionCallAstNode.java]: sorry, it was bad merged. removed changes

File: pinot-common/src/main/java/org/apache/pinot/common/function/AggregationFunctionType.java (hunk 2 above)
Comment (siddharthteotia) [reply pinot-common/src/main/java/org/apache/pinot/common/function/AggregationFunctionType.java]: yes, removed changes. thanks for catching

**Comment by siddharthteotia** [comment]
Got a clean build. Let's merge this.

**Pull request merged**